Packs/VMware/Integrations/VMware/VMware.py

```diff
@@ -373,48 +554,313 @@ def change_nic_state(args):
+        'Type': entryTypes['note'],
+        'Contents': {},
+        'ReadableContentsFormat': formats['text'],
+        'HumanReadable': 'Virtual Machine was unregistered successfully.',
+        'EntryContext': {}
+    }
+
+
+def test_module(si):
+    get_vms(si, {'limit': '1'})
+    return 'ok'
+
+
+def main():  # pragma: no cover
+    sout = sys.stdout
+    sys.stdout = StringIO()
+    res = []
+    si = None
+    try:
+        si, vsphere_client = login(demisto.params())
+
+        if demisto.command() == 'test-module':
+            result = test_module(si)
+        if demisto.command() == 'vmware-get-vms':
+            result = get_vms(si, demisto.args())
+        if demisto.command() == 'vmware-poweron':
+            result = power_on(si, demisto.args()['vm-uuid'])
+        if demisto.command() == 'vmware-poweroff':
+            result = power_off(si, demisto.args()['vm-uuid'])
+        if demisto.command() == 'vmware-hard-reboot':
+            result = hard_reboot(si, demisto.args()['vm-uuid'])
+        if demisto.command() == 'vmware-suspend':
+            result = suspend(si, demisto.args()['vm-uuid'])
+        if demisto.command() == 'vmware-soft-reboot':
+            result = soft_reboot(si, demisto.args()['vm-uuid'])
+        if demisto.command() == 'vmware-create-snapshot':
+            result = create_snapshot(si, demisto.args())
+        if demisto.command() == 'vmware-revert-snapshot':
+            result = revert_snapshot(si, demisto.args()['snapshot-name'], demisto.args()['vm-uuid'])
+        if demisto.command() == 'vmware-get-events':
+            result = get_events(si, demisto.args())
+        if demisto.command() == 'vmware-change-nic-state':
+            result = change_nic_state(si, demisto.args())
+        if demisto.command() == 'vmware-list-vms-by-tag':
+            result = list_vms_by_tag(vsphere_client, demisto.args())
+        if demisto.command() == 'vmware-create-vm':
+            result = create_vm(si, demisto.args())
+        if demisto.command() == 'vmware-clone-vm':
+            result = clone_vm(si, demisto.args())
+        if demisto.command() == 'vmware-relocate-vm':
+            result = relocate_vm(si, demisto.args())
+        if demisto.command() == 'vmware-delete-vm':
+            result = delete_vm(si, demisto.args())
+        if demisto.command() == 'vmware-register-vm':
+            result = register_vm(si, demisto.args())
+        if demisto.command() == 'vmware-unregister-vm':
+            result = unregister_vm(si, demisto.args())
+        res.append(result)
+    except Exception as ex:
+        if hasattr(ex, 'msg') and ex.msg:  # type: ignore
+            message = ex.msg  # type: ignore
+        else:
+            message = ex
+        res.append(
+            {"Type": entryTypes["error"], "ContentsFormat": formats["text"],
+             "Contents": str(message)})  # type: ignore
+
+    try:
+        logout(si)
+    except Exception as ex:
+        res.append({  # type: ignore
```

Comment:
```suggestion
    except Exception as ex:
        demisto.error(traceback.format_exc())  # print the traceback
        res.append({  # type: ignore
```
need to import traceback for this